### PR TITLE
Added tree widget to winforms [WIP]

### PR DESCRIPTION
**Pull request by obulat**
Signed-off-by: obulat <obulat@gmail.com>

This starts adding tree widget to winforms platform. Tree in the demo app works. However, I'm not sure what the tree should look like when there are several accessors. Tree example apps (tree and tree_source) don't work either, they just show blank box. So, I need help figuring out how to fix it.

## PR Checklist:

- [x] All new features have been tested
- [ ] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

### CODE REVIEW

File: src/winforms/toga_winforms/widgets/tree.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    def change_source(self, source):
+
+        self.native.BeginUpdate()
+
+        def create_children(node):
+            if node._children is None or len(node._children) == 0:
+                # FIXME Works if there is only one accessor. Not sure what the behaviour
+                # should be with many accessors
+                return WinForms.TreeNode(getattr(node, self.interface._accessors[0]))
```
Comment (freakboy3742) [inline src/winforms/toga_winforms/widgets/tree.py]: On macOS and GTK, a Tree widget can still have multiple columns. So, you use the tree's +/- button to expand the tree's levels in the first column, but there can be additional columns showing other details. For example, the "Tree Source" example project shows multiple columns of detail for a movie, with the tree allowing you to sort by decade:
screen shot 2018-03-06 at 2 18 28 pm

The accessors then allow you to specify which attribute will be column 0, which will be column 1, and so on. Does that make sense?

Comment (obulat) [reply src/winforms/toga_winforms/widgets/tree.py]: Thanks, this is what I needed. Will be working on it

File: src/core/toga/widgets/icon.py [hunk 2]
```diff
@@ -26,10 +26,12 @@ class Icon:
...
 
     def __init__(self, path, system=False):
-        if os.path.splitext(path)[1] in ('.png', '.icns', '.bmp'):
+        # Only one filename is provided; platform implementation classes check
+        # if file with suitable extension is available.
+        if os.path.splitext(path)[1] in ('.png', '.icns', '.bmp', '.ico'):
             self.path = path
         else:
-            self.path = path + '.icns'
+            self.path = path
```
Comment (freakboy3742) [inline src/core/toga/widgets/icon.py]: Is this `if` clause needed at all any more?

**Comment by obulat** [comment]
This implementation is not correct, and should be done differently.

**Pull request closed without merging**